src/g2.rs

```diff
@@ -827,6 +827,109 @@ impl G2Projective {
         acc
     }
 
+    fn psi(&self) -> G2Projective {
```

Comment:
We should have a test for this specifically which exercises nontrivial z. (Probably given a point, run psi(P) on nontrivial z, and convert the original point to affine and then back to projective and run psi again, then compare?)